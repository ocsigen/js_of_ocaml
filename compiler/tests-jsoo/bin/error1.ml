let () =
  match Array.to_list Sys.argv with
  | _ :: "unregister" :: _ ->
      let null = Array.unsafe_get [| 1 |] 1 in
      Callback.register "Printexc.handle_uncaught_exception" null
  | _ -> ()

exception D of int * string * Int64.t

let _ =
  Printexc.register_printer (function
      | D _ -> Some "custom printer"
      | _ -> None)

let _ = raise (D (2, "test", 43L))
