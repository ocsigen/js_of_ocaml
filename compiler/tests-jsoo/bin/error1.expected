Fatal error: exception custom printer
